### PR TITLE
Add portable installers to Coder 2.4.0

### PR DESCRIPTION
**Pull request by mdanish-kh**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/129175)

### CODE REVIEW

File: manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml [hunk 1]
```diff
@@ -17,6 +17,8 @@ Installers:
   NestedInstallerFiles:
     - RelativeFilePath: coder.exe
       PortableCommandAlias: coder
+  AppsAndFeaturesEntries:
+    - DisplayName: Coder (portable)
```
Comment (matifali) [inline manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml:21]:
```suggestion
    - DisplayName: Coder
```

File: manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml [hunk 2]
```diff
@@ -25,6 +27,8 @@ Installers:
   NestedInstallerFiles:
     - RelativeFilePath: coder.exe
       PortableCommandAlias: coder
+  AppsAndFeaturesEntries:
+    - DisplayName: Coder (portable)
   InstallerUrl: https://github.com/coder/coder/releases/download/v2.4.0/coder_2.4.0_windows_arm64.zip
```
Comment (matifali) [inline manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml:32]:
```
    - DisplayName: Coder
```
Same observation.

File: manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml [hunk 3]
```diff
@@ -25,6 +27,8 @@ Installers:
   NestedInstallerFiles:
     - RelativeFilePath: coder.exe
       PortableCommandAlias: coder
+  AppsAndFeaturesEntries:
+    - DisplayName: Coder (portable)
```
Comment (matifali) [inline manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml:31]:
```suggestion
    - DisplayName: Coder
```
Same observation

File: manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml (hunk 1 above)
Comment (mdanish-kh) [reply manifests/c/Coder/Coder/2.4.0/Coder.Coder.installer.yaml:21]: For portables, we actually control what the DisplayName written to Registry (visible in Control Panel / Apps&Features is). If I don't add this distinction, and user has a side-by-side installation of portable with the nullsoft installer, the user would not get a clear distinction of the installer type in the Control Panel. See -

image

Adding (portable) helps user better identify the package in Control Panel.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**